TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java

```diff
@@ -113,18 +114,26 @@ private void writeObservationTag(int fieldNumber) throws IOException {
                 getWireTypeForFieldIndex(TrafficObservation.getDescriptor(), fieldNumber));
     }
 
-    private void beginSubstreamObservation(Instant timestamp, int captureTag, int captureClosureSize) throws IOException {
-
-        // i.e. 2 {
+    /**
+     * Will write the beginning fields for a TrafficObservation after first checking if sufficient space exists in the
+     * CodedOutputStream and flushing if space does not exist. This should be called before writing any observation to
+     * the TrafficStream.
+     */
+    private void beginSubstreamObservation(Instant timestamp, int captureTagFieldNumber, int captureTagLengthAndContentSize) throws IOException {
+        final var tsContentSize = CodedOutputStreamSizeUtil.getSizeOfTimestamp(timestamp);
+        final var tsTagSize = CodedOutputStream.computeInt32Size(TrafficObservation.TS_FIELD_NUMBER, tsContentSize);
+        final var captureTagNoLengthSize = CodedOutputStream.computeTagSize(captureTagFieldNumber);
+        final var observationContentSize = tsTagSize + tsContentSize + captureTagNoLengthSize + captureTagLengthAndContentSize;
+        // Ensure space is available before starting an observation
+        if (getOrCreateCodedOutputStream().spaceLeft() < CodedOutputStreamSizeUtil.bytesNeededForObservationAndClosingIndex(
+            observationContentSize, numFlushesSoFar + 1)) {
```

Comment:
Can you please reformat this?  This is the hardest to read line break I've seen in a long while.  Maybe break after '<' and/or put the brace on the next line.